Description: Place libraries after archive file to have dependencies resolved
 when --as-needed linker option is being used.
Author: Ilya Barygin <barygin@gmail.com>
Bug-Debian: http://bugs.debian.org/640202

--- hwinfo-16.0.orig/src/Makefile
+++ hwinfo-16.0/src/Makefile
@@ -23,9 +23,10 @@
 
 ifdef SHARED_FLAGS
 $(LIBHD_SO): $(LIBHD)
-	$(CC) -shared $(SO_LIBS) -Wl,--whole-archive $(LIBHD) -Wl,--no-whole-archive \
+	$(CC) -shared -Wl,--whole-archive $(LIBHD) -Wl,--no-whole-archive \
 		-Wl,-soname=$(LIBHD_SONAME) \
-		-o $(LIBHD_SO) 
+		-o $(LIBHD_SO) \
+		$(SO_LIBS)
 	ln -snf $(LIBHD_NAME) $(LIBHD_SONAME)
 	ln -snf $(LIBHD_SONAME) $(LIBHD_BASE).so
 else
